Core/Frameworks/Flake/Framework.php

```diff
@@ -325,6 +327,37 @@ protected static function initDbMysql(array $config) {
         return true;
     }
 
+    protected static function initDbPgsql(array $config) {
+        if (!$config['database']['pgsql_host']) {
+            exit("<h3>The constant PROJECT_DB_PGSQL_HOST, containing the PostgreSQL host name, is not set.<br />You should set it in config/baikal.yaml</h3>");
+        }
+
+        if (!$config['database']['pgsql_dbname']) {
+            exit("<h3>The constant PROJECT_DB_PGSQL_DBNAME, containing the PostgreSQL database name, is not set.<br />You should set it in config/baikal.yaml</h3>");
+        }
+
+        if (!$config['database']['pgsql_username']) {
+            exit("<h3>The constant PROJECT_DB_PGSQL_USERNAME, containing the PostgreSQL database username, is not set.<br />You should set it in config/baikal.yaml</h3>");
+        }
+
+        if ($config['database']['pgsql_password'] === null) {
```

Comment:
I assume the `=== null` part is to permit an empty password, but I ended up getting this error after leaving the password field blank during setup.

For context, I'm using a local connection with peer authentication, and my application user accounts do not have passwords.  A username is also not necessarily required as [PostgreSQL will query the system for it on peer connections](https://www.postgresql.org/docs/current/auth-peer.html).

The setup page did successfully configure the database without either of a username or password specified, so I've just removed this and the username check above on my local setup.

Reply:
Hey there :) Thanks for pointing this out!

I've applied both suggestions (check for `PDO::pgsql` and tolerant checks to allow empty username and password) in be51975, you can see the changes here: [Compare ba64521..be51975](https://github.com/sabre-io/Baikal/compare/ba6452ad2a4853e2903dbe86834940dd02116ee5..a43af0262d90c329e4b53a3ecd74ad52d0b04c28)